### PR TITLE
Some cleanup of ActionResults - #657

### PR DESCRIPTION
**Pull request by rynowak**
In general all properties are get/set so filters can change them.
 - some validate for not-null
 - where we use services it's get/set also

Services are resolved in the Execute method if not provided.

A few more ActionResults that return a body have the ability to set a
status code now (optional).

### CODE REVIEW

**Comment by rynowak** [comment]
@yishaigalatzer

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/CreatedAtActionResult.cs [hunk 1]
```diff
@@ -41,17 +41,17 @@ public class CreatedAtActionResult : ObjectResult
         /// <summary>
         /// Gets the name of the action to use for generating the URL.
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/CreatedAtActionResult.cs]: nit: `Gets` or `sets` comment for all these type of changes

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs [hunk 2]
```diff
@@ -48,7 +50,7 @@ public FilePathResult([NotNull] string fileName, [NotNull] string contentType)
         public FilePathResult(
             [NotNull] string fileName,
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs]: make the property get/set, and remove IFileSystem from the constructor

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/FileResult.cs [hunk 3]
```diff
@@ -31,7 +31,7 @@ protected FileResult([NotNull] string contentType)
         /// <summary>
         /// Gets the Content-Type header value that will be written to the response.
         /// </summary>
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/FileResult.cs]: remove the `[NotNull]` from the constructor

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/FileStreamResult.cs [hunk 4]
```diff
@@ -24,7 +27,7 @@ public class FileStreamResult : FileResult
         /// </summary>
         /// <param name="fileStream">The stream with the file.</param>
         /// <param name="contentType">The Content-Type header of the response.</param>
-        public FileStreamResult([NotNull] Stream fileStream, string contentType)
+        public FileStreamResult([NotNull] Stream fileStream, [NotNull] string contentType)
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/FileStreamResult.cs]: two constructors

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/FileContentResult.cs [hunk 5]
```diff
@@ -13,6 +14,8 @@ namespace Microsoft.AspNet.Mvc
     /// </summary>
     public class FileContentResult : FileResult
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/FileContentResult.cs]: should we add StatusCode property on FileContent and FilePath results?

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/JsonResult.cs [hunk 6]
```diff
@@ -87,6 +93,13 @@ public JsonResult(object value, IOutputFormatter formatter)
             };
 
             var formatter = SelectFormatter(objectResult, formatterContext);
+            Debug.Assert(formatter != null);
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/JsonResult.cs]: lets include a clear comment about this behavior.

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectResult.cs [hunk 7]
```diff
@@ -25,18 +25,18 @@ public RedirectResult(string url, bool permanent)
             Url = url;
         }
 
-        public bool Permanent { get; private set; }
+        public bool Permanent { get; set; }
 
-        public string Url { get; private set; }
+        public string Url { get; set; }
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectResult.cs]: nullcheck

**Comment by yishaigalatzer** [comment]
:shipit: when done with comments

**Pull request closed without merging**